compat/src/render.js

```diff
@@ -114,24 +121,23 @@ options.vnode = vnode => {
 
 	// only normalize props on Element nodes
 	if (typeof type === 'string') {
-		const nonCustomElement = type.indexOf('-') === -1;
 		normalizedProps = {};
 
 		for (let i in props) {
 			let value = props[i];
 
-			if (IS_DOM && i === 'children' && type === 'noscript') {
+			if (
+				(i === 'value' && 'defaultValue' in props && value == null) ||
 				// Emulate React's behavior of not rendering the contents of noscript tags on the client.
-				continue;
-			} else if (i === 'value' && 'defaultValue' in props && value == null) {
+				(IS_DOM && i === 'children' && type === 'noscript')
+			) {
 				// Skip applying value if it is null/undefined and we already set
 				// a default value
 				continue;
-			} else if (
-				i === 'defaultValue' &&
-				'value' in props &&
-				props.value == null
-			) {
+			}
+
+			let lowerCased = i.toLowerCase();
```

Comment:
Ah, is the hypothesis here that doing an upfront lowercase + case-sensitive regex is faster than case-insensitive regex?

Reply:
A part of me wonders if for this it'd be faster to take the size penalty and just do full name comparisons (`i === 'onInput'`, or `i.startsWith('onAnim')`)

Reply:
Probably ye